db/migrate/20220307154148_add_devise_to_users.rb

```diff
@@ -0,0 +1,48 @@
+class AddDeviseToUsers < ActiveRecord::Migration[7.0]
+  def self.up
+    change_table :users do |t|
+      ## Database authenticatable
+      t.string :email, null: false, default: ''
+      t.string :encrypted_password, null: false, default: ''
+
+      ## Recoverable
+      t.string :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Trackable
+      # t.integer  :sign_in_count, default: 0, null: false
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      # Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```

Comment:
- [ ] I saw from your migration files that you created migrations for the user `confirmable` yourself. It actually works 👍  however, it is required to use devise for user authentication. Devise has all this confirmable already figured out for you. All you need to do is to uncomment the lines of code I highlighted so that devise can add everything about user confirmation to the `users` table instead of you doing it yourself. Do not forget to delete all the migrations which you created for the purpose of user confirmation as devise will take care of it. Also do not forget to run migrations and test your app to see the change in action.